subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompiler.java

```diff
@@ -78,6 +78,7 @@ static WorkResult execute(ServiceRegistry serviceRegistry, final Iterable<File>
             }
 
             if (spec.getScalaCompileOptions().isForce()) {
+                // TODO This should use Deleter
```

Comment:
Will this be fixed in a follow up?

Reply:
I added https://github.com/gradle/gradle/issues/10277. This is a bit more complicated as `Deleter` would need to be exposed to workers etc.